MediaBrowser.LocalMetadata/Savers/BaseXmlSaver.cs

```diff
@@ -94,7 +94,7 @@ public async Task SaveAsync(BaseItem item, CancellationToken cancellationToken)
         {
             var path = GetSavePath(item);
             var directory = Path.GetDirectoryName(path) ?? throw new InvalidDataException($"Provided path ({path}) is not valid.");
-            Directory.CreateDirectory(directory);
+            _ = directory.Length > 260 ? Directory.CreateDirectory(directory[..260]) : Directory.CreateDirectory(directory);
```

Comment:
just truncating the path is a poor way to fix this.
For windows you could got with the extended path sytax instead:
https://learn.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation?tabs=registry

Or truncate it in the usual windows way by using the tilde character.

also for linux the max _folder_ and _filename_ length is 255 which will potentially break a lot of systems. The Path length is usually 4000.

Reply:
I see your point, it would make more sense to either use Windows extended path syntax or just cut the folder name short depending on OS. I'll close this PR and open a new one once I implement that.

Reply:
hmm here it gets even worse, usually directory length restrictions are not OS limitations but filesystem ones. Each filesystem has other length limitations.

Reply:
That does complicate things. Would it make sense to query the filesystem using [this property](https://learn.microsoft.com/en-us/dotnet/api/system.io.driveinfo.driveformat?view=net-8.0&redirectedfrom=MSDN#System_IO_DriveInfo_DriveFormat) and then map each filesystem to a NAME_MAX? Or would it be best to just use 255 as a catch-all for Unix systems? Very few (common) filesystems seem to deviate from 255.

Reply:
Well for folder length anyway not path length. I think there is somewhere a variable you can get to get the max folder length yes but not the max path length ... or vice versa.

Reply:
Tbo you could just catch the error and log it as is to not break the flow for other actions in the calling code, but i dont see how that can be fixed without opening a huge hole in the code (as this is literally everywhere in code "an issue")

Reply:
If metadata wasn't automatically pulled then this wouldn't be an issue (or at least much less of one). I don't think just logging it would be an effective solution because of this: automatically pulling metadata is expected to work without user interaction. The majority of Unix-like filesystems don't have a PATH_MAX less than 4096, so I think it might be fair to leave it that way as a catch-all, as with with 255 for NAME_MAX.